packages/icon-button/index.js

```diff
@@ -0,0 +1,129 @@
+import React, {Component} from 'react';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import withRipple from '@material/react-ripple';
+import {MDCIconButtonToggleFoundation} from '@material/icon-button';
+
+const {strings} = MDCIconButtonToggleFoundation;
+
+export class IconButtonBase extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      classList: new Set(),
+      childContent: props.children,
+      [strings.ARIA_PRESSED]: props[strings.ARIA_PRESSED],
+      [strings.ARIA_LABEL]: props[strings.ARIA_LABEL],
+    };
+  }
+
+  componentDidMount() {
+    this.initializeFoundation();
+  }
+
+  initializeFoundation = () => {
+    this.foundation_ = new MDCIconButtonToggleFoundation(this.adapter);
+    this.foundation_.init();
+  }
+
+  get classes() {
+    const {classList} = this.state;
+    const {className} = this.props;
+    return classnames('mdc-icon-button', Array.from(classList), className);
+  }
+
+  get adapter() {
+    return {
+      addClass: (className) =>
+        this.setState({classList: this.state.classList.add(className)}),
+      removeClass: (className) => {
+        const classList = new Set(this.state.classList);
+        classList.delete(className);
+        this.setState({classList});
+      },
+      // the foundation needs data attributes that will be passed in as props if the user wants this feature
```

Comment:
are these the data-toggle-on stuff I saw before?

Reply:
yup this is referring tot he 6 different data attrs